generate_qfp.py

```diff
@@ -0,0 +1,571 @@
+"""
+Generate the following SO packages:
+
+- LQFP / TQFP (only square)
+
+Relevant standards:
+
+- JEDEC MS-026 https://www.jedec.org/system/files/docs/MS-026D.pdf
+
+"""
+from collections import namedtuple
+from copy import deepcopy
+from itertools import chain
+from os import path, makedirs
+from typing import Iterable, List, Optional
+from uuid import uuid4
+
+from common import now, init_cache, save_cache
+from common import format_float as ff, format_ipc_dimension as fd
+from common import COURTYARD_LINE_WIDTH, sign
+
+
+generator = 'librepcb-parts-generator (generate_qfp.py)'
+
+line_width = 0.25
+pkg_text_height = 1.0
+text_y_offset = 1.0
+silkscreen_offset = 0.150  # 150 µm
+
+
+# Initialize UUID cache
+uuid_cache_file = 'uuid_cache_qfp.csv'
+uuid_cache = init_cache(uuid_cache_file)
+
+
+# Excess as a function of pitch according to IPC-7351C.
+Excess = namedtuple('Excess', 'toe heel side courtyard')
+DENSITY_LEVEL_A = {  # Most
+    1.00: Excess(0.40, 0.50,  0.07, 0.40),
+    0.80: Excess(0.35, 0.45,  0.06, 0.40),
+    0.65: Excess(0.30, 0.40,  0.05, 0.40),
+    0.50: Excess(0.25, 0.35,  0.03, 0.40),
+    0.40: Excess(0.20, 0.30,  0.00, 0.40),
+    0.35: Excess(0.20, 0.30, -0.01, 0.40)
+}
+DENSITY_LEVEL_B = {  # Nominal
+    1.00: Excess(0.35, 0.45,  0.06, 0.20),
+    0.80: Excess(0.30, 0.40,  0.05, 0.20),
+    0.65: Excess(0.25, 0.35,  0.04, 0.20),
+    0.50: Excess(0.20, 0.30,  0.02, 0.20),
+    0.40: Excess(0.15, 0.25, -0.01, 0.20),
+    0.35: Excess(0.15, 0.25, -0.02, 0.20)
+}
+DENSITY_LEVEL_C = {  # Least
+    1.00: Excess(0.30, 0.40,  0.05, 0.10),
+    0.80: Excess(0.25, 0.35,  0.04, 0.10),
+    0.65: Excess(0.20, 0.30,  0.03, 0.10),
+    0.50: Excess(0.15, 0.25,  0.01, 0.10),
+    0.40: Excess(0.10, 0.20, -0.02, 0.10),
+    0.35: Excess(0.10, 0.20, -0.03, 0.10)
+}
```

Comment:
Hmm I guess these excess values are not correct, at least not according IPC-7351B:

![grafik](https://user-images.githubusercontent.com/5374821/52919023-2ed05d00-32fe-11e9-98e6-8ac252391d46.png)

Note the `<=` operator ;)

Reply:
Oh, you are right. Well spotted indeed :slightly_smiling_face: